packages/cli/src/services/orchestration/main/MultiMainSetup.ee.ts

```diff
@@ -89,4 +97,32 @@ export class MultiMainInstancePublisher extends SingleMainInstancePublisher {
 			await this.redisPublisher.setExpiration(this.leaderKey, this.leaderKeyTtl);
 		}
 	}
+
+	async broadcastWorkflowWasUpdated(workflowId: string, pushSessionId = '') {
+		if (!this.sanityCheck()) return;
+
+		await this.redisPublisher.publishToCommandChannel({
+			command: 'workflowWasUpdated',
+			payload: { workflowId, pushSessionId },
+		});
+	}
+
+	async broadcastWorkflowWasActivated(workflowId: string, targets: string[], pushSessionId = '') {
+		if (!this.sanityCheck()) return;
+
+		await this.redisPublisher.publishToCommandChannel({
+			command: 'workflowWasActivated',
+			targets,
+			payload: { workflowId, pushSessionId },
+		});
+	}
+
+	async broadcastWorkflowWasDeactivated(workflowId: string, pushSessionId = '') {
+		if (!this.sanityCheck()) return;
+
+		await this.redisPublisher.publishToCommandChannel({
+			command: 'workflowWasDeactivated',
+			payload: { workflowId, pushSessionId },
+		});
+	}
```

Comment:
We could extend the sanityCheck() with more checks, though I'm not sure what we could add. We are currently testing for queue mode and whether it is main, but only the Leader should be sending these out since otherwise we can start an endless loop when the receivers start reactivating theirs...?